### PR TITLE
Refs #17209 -- Added LoginView and LogoutView class-based views

### PR DESCRIPTION
**Pull request by claudep**

### CODE REVIEW

**Comment by auvipy** [comment]
thanks for the better working patch!! isort builds are failing though! I could update the remaining view later

**Comment by phildini** [comment]
Now tests are passing, but the PR needs to be re-rebased, and the references to 1.10 should probably be updated, as we have passed the window for it.

File: docs/topics/auth/default.txt [hunk 1]
```diff
@@ -957,12 +957,31 @@ implementation details see :ref:`using-the-views`.
 
 .. function:: login(request, template_name=`registration/login.html`, redirect_field_name='next', authentication_form=AuthenticationForm, current_app=None, extra_context=None, redirect_authenticated_user=False)
 
+    .. deprecated:: 1.10
```
Comment (auvipy) [inline docs/topics/auth/default.txt]: shouldn't this 1.11?

Comment (claudep) [inline docs/topics/auth/default.txt]: Absolutely, I'll made a second pass to check that.

**Comment by timgraham** [comment]
LGTM. Some cosmetic edits: http://dpaste.com/16DTTB3

**Pull request closed without merging**